extensions/oplogPopulator/OplogPopulator.js

```diff
@@ -49,8 +49,12 @@ class OplogPopulator {
             logger: this._logger,
         });
         this._changeStreamWrapper = null;
-        // stores the set of buckets assigned to connector
-        this._bucketsForConnector = new Set();
+        // contains created connector names
+        this._configuredConnectors = [];
+        // stores connector assigned for each bucket
+        this._connectorToBucket = new Map();
+        // stores sets of buckets assigned to each connector
+        this._bucketsForConnectors = {};
```

Comment:
should be a field inside each `Connector` (i.e. each entry of the `_configuredConnectors` list)